### PR TITLE
Dashboard ide

### PR DESCRIPTION
**Pull request by chdattat**

### CODE REVIEW

**Comment by chdattat** [comment]
Hi @jambulud, @sjimenez77,
Please review the code for the tasks - Deleting a project and opening a project path location.

Thanks,
Dattu Chillal

File: main/index.ts [hunk 1]
```diff
@@ -279,6 +279,28 @@ const openProjectInIde = (project: ProjectDetails) => {
...
+      });
+    })
+    .catch((error) => {
+      console.log(error);
+      mainWindow.webContents.send('delete:project', {
+        projects: [],
+        message: 'error',
+      });
+    });
+};
```
Comment (jambulud) [inline main/index.ts]: Move to another file

File: main/index.ts [hunk 2]
```diff
@@ -279,6 +279,28 @@ const openProjectInIde = (project: ProjectDetails) => {
...
+
+const openProjectDirectory = (path: string) => {
+  shell.showItemInFolder(path);
+};
```
Comment (jambulud) [inline main/index.ts]: Is this function needed? it seems to be used only once

File: main/services/devon-instances/devon-instances.service.ts [hunk 3]
```diff
@@ -226,12 +231,33 @@ export class DevonInstancesService implements SaveDetails {
...
   ): Promise<ProcessState> {
-    return await utilExec(this.findCommand(project.domain), {
-      cwd: project.path,
-    });
+    try {
+      return await utilExec(this.findCommand(project.domain), {
+        cwd: project.path,
+      });
+    } catch (error) {
+      console.log(error);
```
Comment (jambulud) [inline main/services/devon-instances/devon-instances.service.ts]: replace to console.error

**Review by jambulud — CHANGES_REQUESTED** [review]

File: main/index.ts (hunk 2 above)
Comment (chdattat) [reply main/index.ts]: Yes. It's needed. This is the function to open project path location

File: renderer/components/dashboard-projects/menu-list.tsx [hunk 4]
```diff
@@ -19,13 +39,53 @@ export default function MenuList(props: {
...
+            <List className={classes.sublist}>
+              <ListItem button>
+                <div onClick={() => props.openProjectInIde('vscode')}>
+                  In Visual Studio Code
+                </div>
+              </ListItem>
+              {props.project &&
+              props.project.domain !== '' &&
+              props.project.domain !== 'angular' &&
+              props.project.domain !== 'node' ? (
```
Comment (jambulud) [inline renderer/components/dashboard-projects/menu-list.tsx]: replace to
 
```ts
!['angular', 'node'].includes(props.project?.domain)
```
This way you avoid multiple checks. Also the "Optional chaining" helps you accessing a nested property returning undefined if any of the levels file (you could do a?.b?.c?.d?.myProperty)

File: renderer/components/dashboard-projects/project-detail.tsx [hunk 5]
```diff
@@ -3,16 +3,9 @@ import CardContent from '@material-ui/core/CardContent';
...
 export default function ProjectDetail(props: {
   project: ProjectDetails;
-  handleClose: () => void;
-  openProjectInIde: () => void;
-  openProjectDirectory: () => void;
-  deleteProject: () => void;
-  state: ProjectMenuType;
   handleClick: (
     event: React.MouseEvent<HTMLDivElement>,
     project: ProjectDetails
```
Comment (jambulud) [inline renderer/components/dashboard-projects/project-detail.tsx]: Move to a ProjectDetailProps interface (located in the same file)

File: renderer/components/dashboard-projects/menu-list.tsx [hunk 6]
```diff
@@ -1,14 +1,34 @@
...
+import { menuStyles } from './menu-list.styles';
+import { ProjectDetails } from '../../modules/projects/redux/stepper/data.model';
 
 export default function MenuList(props: {
   state: ProjectMenuType;
   handleClose: () => void;
-  openProjectInIde: () => void;
+  openProjectInIde: (ide: string) => void;
   openProjectDirectory: () => void;
   deleteProject: () => void;
```
Comment (jambulud) [inline renderer/components/dashboard-projects/menu-list.tsx]: Move to interface MenuListProps so you can do `MenuList(props: MenuListProps)`

The interface has to be located in the same file

**Review by jambulud — CHANGES_REQUESTED** [review]

File: renderer/components/dashboard-projects/menu-list.tsx (hunk 4 above)
Comment (chdattat) [reply renderer/components/dashboard-projects/menu-list.tsx]: thanks

**Review by jambulud — APPROVED** [review]

**Pull request merged**